bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/zk/ZKMetadataDriverBase.java

```diff
@@ -260,6 +268,46 @@ public synchronized LedgerManagerFactory getLedgerManagerFactory()
         return lmFactory;
     }
 
+    public CompletableFuture<Void> disableHealthCheck() {
+        CompletableFuture<Void> createResult = new CompletableFuture<>();
+        try {
+            zk.create(disableHealthCheckPath, BookKeeperConstants.EMPTY_BYTE_ARRAY, acls, CreateMode.PERSISTENT);
```

Comment:
it is better to use the async API of ZooKeeper, this is why I told you about using CompletableFuture

Reply:
you are right! I will fix!

Reply:
like this?

   public CompletableFuture<Void> disableHealthCheck() {
        CompletableFuture<Void> createResult = new CompletableFuture<>();
        zk.create(disableHealthCheckPath, BookKeeperConstants.EMPTY_BYTE_ARRAY, acls, CreateMode.PERSISTENT, new AsyncCallback.StringCallback() {
            @Override
            public void processResult(int rc, String path, Object ctx, String name) {
                if (KeeperException.Code.OK.intValue() == rc) {
                    createResult.complete(null);
                } else if (KeeperException.Code.NODEEXISTS.intValue() == rc) {
                    log.debug("health check already disable!");
                    createResult.complete(null);
                } else {
                    createResult.completeExceptionally(KeeperException.create(KeeperException.Code.get(rc), path));
                }
            }
        }, null);

        return createResult;
    }

Reply:
fixed method :

`   public CompletableFuture<Void> disableHealthCheck() {
        CompletableFuture<Void> createResult = new CompletableFuture<>();
        zk.create(disableHealthCheckPath, BookKeeperConstants.EMPTY_BYTE_ARRAY, acls, CreateMode.PERSISTENT, new AsyncCallback.StringCallback() {
            @Override
            public void processResult(int rc, String path, Object ctx, String name) {
                if (KeeperException.Code.OK.intValue() == rc) {
                    createResult.complete(null);
                } else if (KeeperException.Code.NODEEXISTS.intValue() == rc) {
                    log.debug("health check already disable!");
                    createResult.complete(null);
                } else {
                    createResult.completeExceptionally(KeeperException.create(KeeperException.Code.get(rc), path));
                }
            }
        }, null);

        return createResult;
    }`

Reply:
like this?

   public CompletableFuture<Void> disableHealthCheck() {
        CompletableFuture<Void> createResult = new CompletableFuture<>();
        zk.create(disableHealthCheckPath, BookKeeperConstants.EMPTY_BYTE_ARRAY, acls,
                CreateMode.PERSISTENT, new AsyncCallback.StringCallback() {
            @Override
            public void processResult(int rc, String path, Object ctx, String name) {
                if (KeeperException.Code.OK.intValue() == rc) {
                    createResult.complete(null);
                } else if (KeeperException.Code.NODEEXISTS.intValue() == rc) {
                    log.debug("health check already disable!");
                    createResult.complete(null);
                } else {
                    createResult.completeExceptionally(KeeperException.create(KeeperException.Code.get(rc), path));
                }
            }
        }, null);

        return createResult;
    }

Reply:
public CompletableFuture<Void> disableHealthCheck() {
        CompletableFuture<Void> createResult = new CompletableFuture<>();
        zk.create(disableHealthCheckPath, BookKeeperConstants.EMPTY_BYTE_ARRAY, acls,
                CreateMode.PERSISTENT, new AsyncCallback.StringCallback() {
        
            public void processResult(int rc, String path, Object ctx, String name) {
                if (KeeperException.Code.OK.intValue() == rc) {
                    createResult.complete(null);
                } else if (KeeperException.Code.NODEEXISTS.intValue() == rc) {
                    log.debug("health check already disable!");
                    createResult.complete(null);
                } else {
                    createResult.completeExceptionally(KeeperException.create(KeeperException.Code.get(rc), path));
                }
            }
        }, null);

        return createResult;
    }